airflow/models/taskinstance.py

```diff
@@ -1385,6 +1385,7 @@ def _prepare_and_execute_task_with_callbacks(self, context, task):
 
         task_copy = task.prepare_for_execution()
         self.task = task_copy
+        context['task'] = self.task
```

Comment:
I'm wondering if it would be better to move those thre lines to L1315 and do:
```py
task_copy = task.prepare_for_execution()
self.task = task_copy
context = self.get_template_context(ignore_param_exceptions=False)
self._prepare_and_execute_task_with_callbacks(context, task)  # Here we probably should pass task_copy?
```
WDYT?

Reply:
And remove _prepare from the name - yeah - would be cleaner.,

Reply:
Yep. Much cleaner !